x-pack/legacy/plugins/siem/public/components/search_bar/index.tsx

```diff
@@ -0,0 +1,326 @@
+ */
+
+import { Filter } from '@kbn/es-query';
+import { getOr, isEqual } from 'lodash/fp';
+import React, { memo, useEffect } from 'react';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+import { Subscription } from 'rxjs';
+import styled from 'styled-components';
+import { StaticIndexPattern, IndexPattern } from 'ui/index_patterns';
+
+import { TimeRange, Query } from 'src/plugins/data/common/types';
+import { SavedQuery } from 'src/legacy/core_plugins/data/public';
+
+import { start as data } from '../../../../../../../src/legacy/core_plugins/data/public/legacy';
+
+import { inputsActions } from '../../store/inputs';
+import { InputsRange } from '../../store/inputs/model';
+import { InputsModelId } from '../../store/inputs/constants';
+import { State, inputsModel } from '../../store';
+import { dispatchUpdateReduxTime, DispatchUpdateReduxTime, formatDate } from '../super_date_picker';
+import {
+  endSelector,
+  filterQuerySelector,
+  fromStrSelector,
+  isLoadingSelector,
+  kindSelector,
+  queriesSelector,
+  savedQuerySelector,
+  startSelector,
+  toStrSelector,
+} from './selectors';
+
+const {
+  ui: { SearchBar },
+  filter,
+  search,
+  timefilter,
+} = data;
+
+export const siemFilterManager = filter.filterManager;
+export const savedQueryService = search.services.savedQueryService;
+
+interface SiemSearchBarRedux {
+  end: number;
+  fromStr: string;
+  isLoading: boolean;
+  queries: inputsModel.GlobalGraphqlQuery[];
+  filterQuery: Query;
+  savedQuery?: SavedQuery;
+  start: number;
+  toStr: string;
+}
+
+interface SiemSearchBarDispatch {
+  updateReduxTime: DispatchUpdateReduxTime;
+  setFilterQuery: ({
+    id,
+    query,
+    language,
+  }: {
+    id: InputsModelId;
+    query: string | { [key: string]: unknown };
+    language: string;
+  }) => void;
+  setSavedQuery: ({
+    id,
+    savedQuery,
+  }: {
+    id: InputsModelId;
+    savedQuery: SavedQuery | undefined;
+  }) => void;
+  setSearchBarFilter: ({ id, filters }: { id: InputsModelId; filters: Filter[] }) => void;
+}
+
+interface SiemSearchBarProps {
+  id: InputsModelId;
+  indexPattern: StaticIndexPattern;
+  timelineId?: string;
+}
+
+const SearchBarContainer = styled.div`
+  .globalQueryBar {
+    padding: 0px;
+  }
+`;
+
+const SearchBarComponent = memo<SiemSearchBarProps & SiemSearchBarRedux & SiemSearchBarDispatch>(
+  ({
+    end,
+    filterQuery,
+    fromStr,
+    id,
+    indexPattern,
+    isLoading = false,
+    queries,
+    savedQuery,
+    setFilterQuery,
+    setSavedQuery,
+    setSearchBarFilter,
+    start,
+    timelineId,
+    toStr,
+    updateReduxTime,
+  }) => {
+    if (fromStr != null && toStr != null) {
+      timefilter.timefilter.setTime({ from: fromStr, to: toStr });
+    } else if (start != null && end != null) {
+      timefilter.timefilter.setTime({
+        from: new Date(start).toISOString(),
+        to: new Date(end).toISOString(),
```

Comment:
Is this where those epochs are sneaking in when loading a saved search that didn't save the timeFilter?

Reply:
No, it was from line [196](https://github.com/elastic/kibana/pull/47306/files/66a6750d48a80f891b980abafeac7bce3f703397#diff-c5a39e1967b16fbccded4e34f033b899R196)